llvm/lib/Target/AIE/AIEBaseAliasAnalysis.cpp

```diff
@@ -77,6 +82,36 @@ static bool isAIEPtrAddIntrinsic(Intrinsic::ID ID, unsigned &InPtrIdx) {
   return false;
 }
 
+/// This gives all indexes for read only operands ie.
+/// excluding counters.
+static SmallVector<unsigned, 5> getAddIntrinsicOps(Intrinsic::ID ID) {
```

Comment:
I think these should be virtual in AIEBaseInstrInfo

Reply:
It could be, however, as in the standard value tracking and basic AA, specific machine instructions are not used, so there is no conflicting opcodes (all instrinsics chare the same namespace). Also, AA is handled mostly in IR level, to the injection of InstrInfo dependency is not trivial here.